R/repair-names.R

```diff
@@ -18,33 +53,77 @@
 #' set_tidy_names(mtcars)
 #'
 #' # By default, all rename operations are printed to the console:
-#' tbl <- as_tibble(structure(list(3, 4, 5), class = "data.frame"), .tidy_names = FALSE)
-#' set_tidy_names(tbl)
+#' tbl <- as_tibble(structure(list(3, 4, 5), class = "data.frame"), .name_repair = "none")
+#' set_valid_names(tbl)
 #'
-#' # Alternatively, use tidy_names() to assign the result manually:
-#' new_names <- tidy_names(names(tbl))
+#' # Alternatively, use valid_names() to assign the result manually:
+#' new_names <- valid_names(names(tbl))
 #' rlang::set_names(tbl, new_names)
 #'
 #' # Optionally, names can be made syntactic:
-#' tidy_names("a b", syntactic = TRUE)
+#' tidy_names("a b")
+#' @name name-repair
+NULL
+
 #' @export
-#' @rdname tidy_names
-set_tidy_names <- function(x, syntactic = FALSE, quiet = FALSE) {
-  new_names <- tidy_names(names2(x), syntactic, quiet)
-  set_names(x, new_names)
+#' @rdname name-repair
+rationalize_names <- function(x, .name_repair) {
+  .name_repair <- .name_repair %||% "none_passive"
+
+  x <- set_minimal_names(x)
+
+  if (is_function(.name_repair)) {
+    repair_fun <- .name_repair
+  } else {
+    repair_fun <- switch(
```

Comment:
The error message isn't helpful if `.name_repair` is not a string, see also https://github.com/r-lib/rlang/issues/589.

```r
as_tibble(iris, .name_repair = TRUE)
```